### PR TITLE
chore(STAK-580): require explicit Metal & Type in Add Inventory modal

### PR DESCRIPTION
**Pull request by lbruton**
## Summary

Implements [STAK-580](https://github.com/lbruton/StakTrakr/blob/dev/.. /DocVault/Projects/StakTrakr/Issues/STAK-580.md) per its Discovery brief — eliminates wrong-metal saves in the Add Inventory Item modal by forcing explicit Metal/Type selection and reordering identity fields above descriptive ones.

User report driving this work: *"the number of times they have added a gold american silver eagle by mistake, being forced to pick the dropdowns before submit, would be a better solution"*

### HTML (`index.html`)
- `#itemMetal` and `#itemType` get `required` plus `<option value="" disabled selected hidden>Select metal…/Select type…</option>` placeholders.
- DOM swap: Metal/Type row now sits **above** Name/Year. Identity fields lead so users declare what kind of item this is before describing it.

### JS (`js/events.js`)
- `filterTypesByMetal` skips the placeholder option in its loop and gates the auto-bump-to-Coin fallback on `selectedOption.value !== ""`. Without these guards, the cascade would re-enable the disabled placeholder and silently coerce Type from placeholder → Coin the moment a user picks a Metal — defeating required Type validation.
- `parseItemFormFields` captures `_rawMetal` / `_rawType` / `_isEditing` before `parseNumistaMetal` coerces empty Metal → `"Alloy"` (which would otherwise pass the existing `!f.metal` truthy gate).
- `validateItemFields` adds an early reject on empty raw Metal/Type for non-edit paths, with friendly message. This backstops `requestSubmit()` (clone-mode save) and any future programmatic submit that bypasses HTML5 native validation.
- `newItemBtn` Add-mode reset adds explicit `itemMetal.value = ""` / `itemType.value = ""` after `form.reset()` as defense-in-depth.
- `validateItemFields` exported on `window` for Playwright.

### Latent `<select>` form-state persistence bug

Resolved structurally. The `selected` attribute on the placeholder option is what makes `form.reset()` return the select to its initial state across modal open/close cycles — that was missing before, which is why a preview build during STAK-576 showed Metal=Gold even though the source had Silver first with no `selected` marker.

### Tests

- New `tests/playwright/stak-580-required-metal-type.spec.js` — 6 tests covering empty placeholders, native validity, JS validator rejection, reset across modal cycles, edit prefill, and the placeholder-state preservation regression.
- `tests/playwright/modal-layout.spec.js` test 2.1 inverted — STAK-559's row-order AC is superseded.
- `tests/playwright/02-crud/crud.spec.js` tests 2.5 + 2.6 updated for required Type (Goldback flow now picks Metal=Gold + Type=Goldback per STAK-562; weight-unit-only sub-adds now explicit Type=Coin).
- `tests/runbook/02-crud.md` parallel updates. Added file-level `<!-- markdownlint-disable MD001 -->` with a note — `# Section` → `### Test` is the runbook-wide convention used by the bb-test parser; lint-staged only saw it when this file was actually modified.

## Verification

- `npx playwright test --reporter=line` → **355 passed** (zero regressions; previous run before fixes had 2 failures: STAK-559 modal-layout 2.1 + crud 2.5 Goldback, both addressed).
- Pre-commit hooks all pass: secrets scan, signing key, sw-cache stamp, prettier, markdownlint.

## Test plan

- [ ] Open the Add Inventory modal — both Metal and Type show their placeholder text and the form does not submit until both are picked.
- [ ] Pick Metal=Gold — Type dropdown still shows the placeholder; Goldback becomes selectable, Silverback hides.
- [ ] Submit with Metal/Type empty — browser shows native "please select an item from the list" tooltip.
- [ ] Save a Gold Goldback. Save a Silver Coin. Save a Platinum Round. All persist correctly with the right metal.
- [ ] Open an existing item via Edit — Metal and Type pre-select with the item's saved values (no placeholder shown).
- [ ] Close and reopen the Add modal — both dropdowns return to their placeholder, no leak from a prior cycle.
- [ ] Clone an existing item via the Save & Clone Another flow — Metal/Type carry over from the source item, save succeeds.

## Notes

- No version bump — rolls into next release.
- Builds on STAK-562 (`TYPE_METAL_FILTER` cascade) and supersedes STAK-559's row-order AC.
- Out of scope (deferred): metal-aware purity defaults (Gold→.9999, Silver→.999, 22K coin→.9167). Today's broad single default of `.9999` (PR #1026) is the current accepted state.

## Release Notes

* **New Features**
  * Metal and Type fields are now required selections in the item form.
  * Metal and Type inputs moved to appear before Name/Year section.
  * Added placeholder prompts for Metal and Type dropdowns to guide user selection.

* **Bug Fixes**
  * Improved form validation to properly detect and prevent empty Metal or Type submissions.
  * Fixed dropdown behavior to preserve placeholder state instead of auto-selecting default options.

### CODE REVIEW

File: tests/runbook/02-crud.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+<!-- markdownlint-disable MD001 -->
+<!-- All runbook files use h1 (# Section) directly followed by h3 (### Test X.Y);
+     test sections are intentionally skipped from h2 so bb-test parsers can match
+     ### headings as test boundaries. Repo-wide convention, not a per-file choice. -->
+
 # Section 02 — CRUD
```
[the remark this answers is not in the document]
Comment (lbruton) [reply tests/runbook/02-crud.md]: **Fixed in `a9d5ecce`** — moved `<!-- markdownlint-disable MD001 -->` (and its explanatory block) to after the `# Section 02 — CRUD` H1, satisfying MD041.

File: js/events.js [hunk 2]
```diff
@@ -1520,18 +1537,25 @@ const filterTypesByMetal = (metalValue) => {
   if (!typeSelect || typeof TYPE_METAL_FILTER === "undefined") return;
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/events.js:1537]: **Fixed in `a9d5ecce`** — replaced `document.getElementById("itemType")` with `safeGetElement("itemType")` in `filterTypesByMetal`. The existing null-guard on the next line already covers the missing-element case.

File: tests/runbook/02-crud.md [hunk 3]
```diff
@@ -90,11 +95,13 @@ _Added: v3.33.25 (STAK-396)_
...
 **Preconditions:** Test 2.4 complete (count = 12). No filters or search active.
 **Steps:**
 
 - act: "click the Add Item button"
-- act: "select Goldback from the metal dropdown"
+- act: "select Gold from the metal dropdown"
+- act: "select Goldback from the type dropdown"
 - act: "type 'BB-GOLDBACK' into the name field"
 - act: "type '1' into the weight field"
 - act: "type '5' into the purchase price field"
```
[the remark this answers is not in the document]
Comment (lbruton) [reply tests/runbook/02-crud.md:107]: **Fixed in `a9d5ecce`** — updated the runbook step to `select '1 Goldback' from the denomination dropdown (#itemGbDenom)`. When Goldback type is selected, the UI hides the weight input and shows the denomination select; the previous step was stale.

File: tests/runbook/02-crud.md [hunk 4]
```diff
@@ -109,32 +116,37 @@ _Added: v3.33.25 (STAK-396)_
 ### Test 2.6 — Add items with each weight unit (oz, g, kg, dwt)
 
 _Added: v3.33.25 (STAK-396)_
+_Updated: STAK-580 — explicit Coin Type selection added; Metal/Type are now required selects with no default._
```
[the remark this answers is not in the document]
Comment (lbruton) [reply tests/runbook/02-crud.md]: **Fixed in `a9d5ecce`** — removed all `dwt` references from Test 2.6 and the cleanup step. Updated heading to `(oz, g, kg, lb)`, item name to `BB-LB-ITEM`, unit selection to `lb`, and weight value to `0.0686` (≈1 troy oz in pounds).

File: js/events.js [hunk 5]
```diff
@@ -1185,6 +1194,14 @@ const parseNumistaDataFields = (isEditing, existingItem, catalog = "") => {
  * @returns {string|null} Error message or null if valid
  */
 const validateItemFields = (f) => {
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/events.js:1196]: **Deferred** — the complexity increase (+24) is inherent to the new Add-mode validation logic and the `filterTypesByMetal` refactor. Unit tests for the raw-value/Alloy coercion edge cases are a valid follow-up; will be tracked separately.

File: js/events.js [hunk 6]
```diff
@@ -1185,6 +1194,14 @@ const parseNumistaDataFields = (isEditing, existingItem, catalog = "") => {
...
+  // STAK-580: explicit check on raw select values for the Add path. The downstream
+  // `!f.metal` gate is bypassed by parseNumistaMetal coercing "" -> "Alloy"; the
+  // raw check closes that hole and also covers `requestSubmit()` flows that skip
+  // HTML5 native validation. Edit path stays permissive — editItem populates the
+  // selects from existing data so this should never trigger there.
+  if (!f._isEditing && (!f._rawMetal || !f._rawType)) {
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/events.js:1202]: **Deferred** — STAK-580 scope is Add mode only (requiring explicit Metal & Type on first save). In Edit mode, existing items always have Metal/Type populated; a user cannot reach the empty-field state through the normal edit flow. Edit-mode validation hardening is a valid but separate concern; will be tracked as a follow-up.
